### PR TITLE
Bolt Cutting and Airlock Examines

### PR DESCRIPTION
**Pull request by Geevies**
Ports https://github.com/Baystation12/Baystation12/pull/15352 slightly.

Adds examining to see if a door's bolt cover is open, and if it has a maglock on it.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by alsoandanswer** [comment]
Why not port it fully?

**Comment by Geevies** [comment]
Despite our best attempts, we are not Bay. This code is a bit more catered to the Auroran playstyle and game balance. I just copied some of the code to make my life easier, and wished to credit the person I retrieved it from.

**Comment by alsoandanswer** [comment]
What were the parts left out though I'm curious

**Comment by Geevies** [comment]
Feel free to take a look: https://github.com/Baystation12/Baystation12/pull/15352/files

Mostly I changed it to take longer and work in stages instead of just being one big thing.

**Comment by Kaedwuff** [comment]
But now how will I, as an evil AI, exploit impenetrable doors by bolting and shutting down the power repeatedly?

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -814,6 +819,91 @@ About the new airlock wires panel:
...
+	else if(istype(tool,/obj/item/weapon/circular_saw))
+		cut_verb = "sawing"
+		cut_sound = 'sound/weapons/saw/circsawhit.ogg'
+		cut_delay *= 2
+		cutting = TRUE
+	else if(istype(tool,/obj/item/weapon/material/twohanded/fireaxe))
+		//fireaxe can smash open the bolt cover instantly
+		var/obj/item/weapon/material/twohanded/fireaxe/F = tool
+		if (!F.wielded)
+			return FALSE
```
Comment (NonQueueingMatt) [inline code/game/machinery/doors/airlock.dm]: span class

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -814,6 +819,91 @@ About the new airlock wires panel:
...
+		if(src.bolt_cut_state == BOLTS_FINE)
+			to_chat(user, "You smash the bolt cover open!")
```
Comment (NonQueueingMatt) [inline code/game/machinery/doors/airlock.dm]: I meant this one

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -814,6 +819,91 @@ About the new airlock wires panel:
...
+	var/cut_verb
+	var/cut_sound
+	var/cutting = FALSE
+
+	if(istype(tool,/obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = tool
+		if(!WT.isOn())
+			return
+		if(!WT.remove_fuel(0,user))
+			to_chat(user, "<span class='notice'>You need more welding fuel to complete this task.</span>")
```
Comment (Geevies) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
			to_chat(user, span("notice", "You need more welding fuel to complete this task."))
```

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (Geevies) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
			to_chat(user, span("warning", "You smash the bolt cover open!"))
```

**Pull request merged**